KarhooSDKTests/Mocks/ObjectTestFactory.swift

```diff
@@ -13,10 +13,17 @@ import CoreLocation
 
 class ObjectTestFactory {
 
-    class func getRandomCredentials(expiryDate: Date? = TestUtil.getRandomDate(),
-                                    withRefreshToken refreshToken: Bool = true) -> Credentials {
-        return Credentials(accessToken: TestUtil.getRandomString(),
-                           expiryDate: expiryDate,
-                           refreshToken: (refreshToken ? TestUtil.getRandomString() : nil))
+    class func getRandomCredentials(
+        expiryDate: Date? = TestUtil.getRandomDate(),
+        withRefreshToken refreshToken: Bool = true,
+        refreshTokenExpiryDate: Date? = TestUtil.getRandomDate()
+    
+    ) -> Credentials {
+        return Credentials(
+            accessToken: TestUtil.getRandomString(),
+            expiryDate: expiryDate,
+            refreshToken: (refreshToken ? TestUtil.getRandomString() : nil),
+            refreshTokenExpiryDate: refreshTokenExpiryDate ?? expiryDate
```

Comment:
shouldn't we take care to have refreshTokenExpiryDate > expiryDate

Reply:
if that's not the case, we'll fallback to the new flow

Reply:
it could produce "random" results in tests, don't you think?

Reply:
cleaned it up